sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/plans/logical/basicLogicalOperators.scala

```diff
@@ -466,6 +488,8 @@ case class View(
 
   override def output: Seq[Attribute] = child.output
 
+  override def metadataOutput: Seq[Attribute] = child.output
```

Comment:
For boundary nodes like `View` and `SubqueryAlias`, I think we should not propagate as the query under them should not change during analysis after they are resolved. `ResolveMissingReferences` skips `SubqueryAlias` as well.